### PR TITLE
Gateway refactoring (NO NOT MERGE)

### PR DESCRIPTION
**Pull request by henrikekblad**
Work in progress...

### CODE REVIEW

File: libraries/MySensors/MySensor.cpp [hunk 1]
```diff
@@ -18,8 +18,29 @@
  */
 
 
+
 #include "MySensor.h"
 
+
```
Comment (henrikekblad) [inline libraries/MySensors/MySensor.cpp]: Below is where the "magic" happens. The .cpp includes will be optioned by #defines created in sketch...

**Comment by henrikekblad** [comment]
Jenkins, build this please.

**Comment by henrikekblad** [comment]
[enable ci]

**Comment by henrikekblad** [comment]
Jenkins, build this please.

**Comment by henrikekblad** [comment]
Jenkins, build this please.

**Comment by fallberg** [comment]
Jenkins, build this please

**Pull request merged**